Mathlib/CategoryTheory/FiberedCategory/HomLift.lean

```diff
@@ -0,0 +1,246 @@
+  mp := fun hφ' => by subst h; simpa using hφ'
+  mpr := fun hφ => inferInstance
+
+@[simp]
+lemma eqToHom_comp_lift_iff {R S : 𝒮} {a b b' : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) (h : b = b') :
+    p.IsHomLift f (φ ≫ eqToHom h) ↔ p.IsHomLift f φ where
+  mp := fun hφ' => by subst h; simpa using hφ'
+  mpr := fun hφ => inferInstance
+
+@[simp]
+lemma lift_eqToHom_comp_iff {R' R S : 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) (h : R' = R) :
+    p.IsHomLift (eqToHom h ≫ f) φ ↔ p.IsHomLift f φ where
+  mp := fun hφ' => by subst h; simpa using hφ'
+  mpr := fun hφ => inferInstance
+
+@[simp]
+lemma lift_comp_eqToHom_iff {R S S' : 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) (h : S = S') :
+    p.IsHomLift (f ≫ eqToHom h) φ ↔ p.IsHomLift f φ where
+  mp := fun hφ' => by subst h; simpa using hφ'
+  mpr := fun hφ => inferInstance
+
+/-- Given a morphism `f : R ⟶ S`, and an isomorphism `φ : a ≅ b` lifting `f`, `isoOfIsoLift f φ` is
+the isomorphism `Φ : R ≅ S` with `Φ.hom = f` induced from `φ` -/
+@[simps hom]
+def isoOfIsoLift  {R S : 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ≅ b) [p.IsHomLift f φ.hom] :
+    R ≅ S where
+  hom := f
+  inv := eqToHom (codomain_eq p f φ.hom).symm ≫ (p.mapIso φ).inv ≫ eqToHom (domain_eq p f φ.hom)
+  hom_inv_id := by subst_hom_lift p f φ.hom; simp [← p.map_comp]
+  inv_hom_id := by subst_hom_lift p f φ.hom; simp [← p.map_comp]
+
+@[simp]
+lemma isoOfIsoLift_inv_hom_id {R S : 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ≅ b) [p.IsHomLift f φ.hom] :
+    (isoOfIsoLift p f φ).inv ≫ f = 𝟙 S :=
+  (isoOfIsoLift p f φ).inv_hom_id
+
+@[simp]
+lemma isoOfIsoLift_hom_inv_id {R S : 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ≅ b) [p.IsHomLift f φ.hom] :
+    f ≫ (isoOfIsoLift p f φ).inv = 𝟙 R :=
+  (isoOfIsoLift p f φ).hom_inv_id
+
+/-- If `φ : a ⟶ b` lifts `f : R ⟶ S` and `φ` is an isomorphism, then so is `f`. -/
+lemma isIso_of_lift_isIso {R S : 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) [p.IsHomLift f φ]
+    [IsIso φ] : IsIso f :=
+  (fac p f φ) ▸ inferInstance
+
+/-- Given `φ : a ≅ b` and `f : R ≅ S`, such that `φ.hom` lifts `f.hom`, then `φ.inv` lifts
+`f.inv`. -/
+protected instance inv_lift_inv {R S : 𝒮} {a b : 𝒳} (f : R ≅ S) (φ : a ≅ b)
+    [p.IsHomLift f.hom φ.hom] : p.IsHomLift f.inv φ.inv := by
+  apply of_commSq
+  apply CommSq.horiz_inv (f:=p.mapIso φ) (commSq p f.hom φ.hom)
+
+/-- Given `φ : a ≅ b` and `f : R ⟶ S`, such that `φ.hom` lifts `f`, then `φ.inv` lifts the
+inverse of `f` given by `isoOfIsoLift`. -/
+protected instance inv_lift {R S : 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ≅ b) [p.IsHomLift f φ.hom] :
+    p.IsHomLift (isoOfIsoLift p f φ).inv φ.inv := by
+  apply of_commSq
+  apply CommSq.horiz_inv (f:=p.mapIso φ) (by simpa using (commSq p f φ.hom))
+
+/-- If `φ : a ⟶ b` lifts `f : R ⟶ S` and both are isomorphisms, then `φ⁻¹` lifts `f⁻¹`. -/
+protected instance inv {R S : 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) [IsIso f] [IsIso φ]
+    [p.IsHomLift f φ] : p.IsHomLift (inv f) (inv φ) :=
+  @IsHomLift.inv_lift_inv _ _ _ _ p _ _ _ _ (asIso f) (asIso φ) (by aesop)
```

Comment:
```suggestion
  have : p.IsHomLift (asIso f).hom (asIso φ).hom := by simp_all
  IsHomLift.inv_lift_inv p (asIso f) (asIso φ)
```
My instinct is to avoid all the @ nonsense.